internal/exporter/prometheus/collector/platform_collector.go

```diff
@@ -0,0 +1,103 @@
+// SPDX-FileCopyrightText: 2025 The Kepler Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package collector
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/sustainable-computing-io/kepler/internal/platform/redfish"
+)
+
+const (
+	// Prometheus namespace for Kepler platform metrics
+	platformSubsystem = "platform"
+)
+
+// RedfishDataProvider defines the interface for getting platform power data
+type RedfishDataProvider interface {
+	ChassisPower() (*redfish.PowerReading, error) // On-demand method for all chassis
+	NodeID() string                               // Node identifier
+	BMCID() string                                // BMC identifier
+}
+
+// PlatformCollector collects platform power metrics from Redfish BMC
+type PlatformCollector struct {
+	logger  *slog.Logger
+	redfish RedfishDataProvider
+
+	// Static metadata
+	nodeID string // Node identifier
+	bmcID  string // BMC identifier
+
+	// Metric descriptors
+	wattsDesc *prometheus.Desc
+}
+
+// NewRedfishCollector creates a new platform collector
+func NewRedfishCollector(redfish RedfishDataProvider, logger *slog.Logger) *PlatformCollector {
+	if redfish == nil {
+		panic("RedfishDataProvider cannot be nil - platform collector requires a data provider to function")
+	}
+	if logger == nil {
+		logger = slog.Default()
+	}
+
+	return &PlatformCollector{
+		logger:  logger,
+		redfish: redfish,
+		nodeID:  redfish.NodeID(),
+		bmcID:   redfish.BMCID(),
+		wattsDesc: prometheus.NewDesc(
+			prometheus.BuildFQName(keplerNS, platformSubsystem, "watts"),
+			"Current platform power consumption in watts from BMC",
+			[]string{"source", "node_name", "bmc", "chassis_id"},
```

Comment:
can follow same convention about `_name` , `_id` , no suffix

Reply:
the labels variables are
```
labels := []string{"redfish", c.nodeID, c.bmcID, reading.ChassisID}
```

the config has field 
```
NodeID      string        `yaml:"nodeID"`
```

Reply:
Right! lets use node_name / nodeName since it follows existing convention for k8s and metrics... 
For Chassis, lets keep the convention .. ChassisID / chassis_id as in https://pkg.go.dev/github.com/stmcginnis/gofish@v0.20.0/common#Entity chassis.ID()